### PR TITLE
docs: section about contract negotiations

### PR DESCRIPTION
**Pull request by wolf4ood**
## What this PR changes/adds

docs: section about contract negotiations

## Why it does that

improve documentation

### CODE REVIEW

File: developer/wip/for-contributors/contributor-handbook.md [hunk 1]
```diff
@@ -580,7 +583,80 @@ The sample expresses that a set of assets identified by their ID be made availab
...
+
+As a side note it is also important to note that contract offers are _ephemeral_ objects as they are generated
+on-the-fly for a particular participant, and they are never persisted in a database and thus cannot be queried through
+any API.
+
+Contract negotiations are asynchronous in nature. That means after initiating them, they become (potentially
+long-running) stateful processes that are advanced by an internal [state machine](#221-state-machines).
+The current state of the negotiation can be queried and altered through the management API.
+
+> In the current iteration all contract offers are automatically accepted by both parties if the contract
```
Comment (paullatzelsperger) [inline developer/wip/for-contributors/contributor-handbook.md:607]: we already have them -> guards: https://github.com/eclipse-edc/Connector/blob/de80081d547c66f127dd32e1a8b27d9fca2d4410/docs/developer/state-machine.md#guards

File: developer/wip/for-contributors/contributor-handbook.md [hunk 2]
```diff
@@ -580,7 +583,80 @@ The sample expresses that a set of assets identified by their ID be made availab
...
+            "authCodeId": "auth-code-id"
+        }
+    ]
+}
+```
+The `counterPartyAddress` is the address where to send the `ContractRequestMessage` via the specified `protocol` (currently [`dataspace-protocol-http`](#27-protocol-extensions-dsp))
+
+The `policy` should hold the same policy associated to the data offering chosen from the [catalog](#216-catalog), plus two additional properties:
+
+- `assigner` the provider ID
```
Comment (paullatzelsperger) [inline developer/wip/for-contributors/contributor-handbook.md:652]:
```suggestion
- `assigner` the providers `participantId`
```

File: developer/wip/for-contributors/contributor-handbook.md [hunk 3]
```diff
@@ -580,7 +583,80 @@ The sample expresses that a set of assets identified by their ID be made availab
...
+- `target` the asset (dataset) ID
+
+In addition the `callbackAddresses`, which is optional, can be used to get notified about state changes of the negotiation. Read more on callbacks in the section about [events and callbacks](#51-events-and-callbacks).
```
Comment (paullatzelsperger) [inline developer/wip/for-contributors/contributor-handbook.md:655]:
```suggestion
In addition, the (optional) `callbackAddresses` array can be used to get notified about state changes of the negotiation. Read more on callbacks in the section about [events and callbacks](#51-events-and-callbacks).
```

**Review by paullatzelsperger — COMMENTED** [review]
small nits, otherwise good to go

**Review by ndr-brt — APPROVED** [review]

**Pull request merged**